src/main.ts

```diff
@@ -302,110 +324,164 @@ async function getTsServerRepoResult(
             }
         }
 
-        const owner = repo.owner ? `${mdEscape(repo.owner)}/` : "";
-        const url = repo.url ? `(${repo.url})` : "";
+        const tsServerResult = {
+            oldServerFailed,
+            oldSpawnResult,
+            newServerFailed,
+            newSpawnResult,
+            replayScriptPath,
+            installCommands,
+        };
 
-        let summary = `## [${owner}${mdEscape(repo.name)}]${url}\n`;
+        if (oldServerFailed && !newServerFailed) {
+            return { status: "Detected interesting changes", tsServerResult }
+        }
+        if (!newServerFailed) {
+            return { status: "Detected no interesting changes" };
+        }
 
-        if (oldServerFailed) {
-            const oldServerError = oldSpawnResult?.stdout
-                ? prettyPrintServerHarnessOutput(oldSpawnResult.stdout, /*filter*/ true)
-                : `Timed out after ${executionTimeout} ms`;
-            summary += `
+        return { status: "Detected interesting changes", tsServerResult, replayScriptPath, rawErrorPath };
+    }
+    catch (err) {
+        reportError(err, `Error running tsserver on ${repo.url ?? repo.name}`);
+        return { status: "Unknown failure" };
+    }
+    finally {
+        console.log(`Done ${repo.url ?? repo.name}`);
+        logStepTime(diagnosticOutput, repo, "language service", lsStart);
+    }
+}
+
+function groupOldErrors(summaries: Summary[]): Map<string, Summary[]> {
+    const group = new Map<string, Summary[]>();
+    for (const summary of summaries) {
+        if (summary.tsServerResult.oldServerFailed && !summary.tsServerResult.newServerFailed) {
+            const error = parseServerHarnessOutput(summary.tsServerResult.oldSpawnResult!.stdout);
+
+            const key = typeof error === "string" ? getHash([error]) : getHashForStack(error.message);
+            const value = group.get(key) ?? [];
+            value.push(summary);
+            group.set(key, value);
+        }
+    }
+
+    return group;
+}
```

Comment:
You can do a single pass and return both old/new errors.

```suggestion
function groupNewAndOldErrors(summaries: Summary[]) {
    const newErrorsGroup = new Map<string, Summary[]>();
    const oldErrorsGroup = new Map<string, Summary[]>();

    for (const summary of summaries) {
        const tsServerResult = summary.tsServerResult;

        let group: Map<string, Summary[]>;
        let spawnResult;
        if (tsServerResult.newServerFailed) {
            group = newErrorsGroup;
            spawnResult = tsServerResult.newSpawnResult;
        }
        else if (tsServerResult.oldServerFailed) {
            group = oldErrorsGroup;
            spawnResult = tsServerResult.oldSpawnResult;
        }
        else {
            continue;
        }

        const error = parseServerHarnessOutput(spawnResult!.stdout);
        const key = typeof error === "string" ? getHash([error]) : getHashForStack(error.message);
        const value = group.get(key) ?? [];
        value.push(summary);
        group.set(key, value);
    }
    
    return { newErrorsGroup, oldErrorsGroup };
}
```

Reply:
That is, if you really want to.

Reply:
Makes sense to me. Accepted.